### PR TITLE
Report DeploymentInPRogress before update manifest signature valid

### PR DESCRIPTION
**Pull request by jw-msft**
- Refactor workflow_parse into smaller, single-purpose functions
- Add workflow_parse_peek_unprotected_workflow_properties to allow OrchestratorCallback to get workflow id and action
- Report DeploymentInProgress before update manifest signature is validated
- Add some stub function placeholders for downloading, verifying/installing rootkey package.

### CODE REVIEW

**Review by nihemstr — APPROVED** [review]
Approved pending Nox

File: src/utils/workflow_utils/src/workflow_utils.c [hunk 1]
```diff
@@ -2157,35 +2408,56 @@ ADUC_Result _workflow_init_helper(ADUC_WorkflowHandle* handle)
...
+    JSON_Value* rootJsonValue = NULL;
+
+    if (updateManifestFile == NULL || outWorkflowHandle == NULL)
     {
         result.ExtendedResultCode = ADUC_ERC_UTILITIES_WORKFLOW_UTIL_ERROR_BAD_PARAM;
         goto done;
     }
 
-    result = _workflow_parse(true, updateManifestFile, validateManifest, handle);
+    result = workflow_parse_json(true /* isFile */, updateManifestFile, &rootJsonValue);
```
Comment (Nox-MSFT) [inline src/utils/workflow_utils/src/workflow_utils.c:2426]: Should set *outWorkflowHandle = NULL before line 2426

**Review by Nox-MSFT — CHANGES_REQUESTED** [review]
Looks good overall. Just minor feedback.

File: src/utils/workflow_utils/src/workflow_utils.c (hunk 1 above)
Comment (jw-msft) [reply src/utils/workflow_utils/src/workflow_utils.c:2428]: fixed

**Review by Nox-MSFT — APPROVED** [review]

**Pull request merged**